tests/plugins/test_custom_plugin.py

```diff
@@ -39,3 +44,54 @@ def test_sync_batchnorm_set(tmpdir):
     )
     trainer.fit(model)
     assert plugin.sync_batchnorm is True
+
+
+@RunIf(skip_windows=True, min_torch="1.7.0")
+def test_ddp_fp16_compress_comm_hook(tmpdir):
+    """Test for DDP FP16 compress hook."""
+    model = BoringModel()
+    plugin = DDPPlugin(
+        ddp_comm_hook=default.fp16_compress_hook,
+    )
+    trainer = Trainer(
+        max_epochs=1,
+        plugins=[plugin],
+        default_root_dir=tmpdir,
+        sync_batchnorm=True,
+    )
+    trainer.fit(model)
+
+
+@RunIf(skip_windows=True, min_torch="1.7.0")
+def test_ddp_sgd_graident_comm_hook(tmpdir):
+    """Test for DDP SGD hook."""
+    model = BoringModel()
+    plugin = DDPPlugin(
+        ddp_comm_state=powerSGD.PowerSGDState(1),
+        ddp_comm_hook=powerSGD.powerSGD_hook,
+    )
+    trainer = Trainer(
+        max_epochs=1,
+        plugins=[plugin],
+        default_root_dir=tmpdir,
+        sync_batchnorm=True,
+    )
+    trainer.fit(model)
+
+
+@RunIf(skip_windows=True, min_torch="1.9.0")
+def test_ddp_fp16_compress_wrapper_comm_hook(tmpdir):
+    """Test for DDP fp16 compress wrapper for SGD hook."""
+    model = BoringModel()
+    plugin = DDPPlugin(
+        ddp_comm_state=powerSGD.PowerSGDState(1),
+        ddp_comm_hook=powerSGD.powerSGD_hook,
+        ddp_comm_wrapper=default.fp16_compress_wrapper,
+    )
+    trainer = Trainer(
+        max_epochs=1,
+        plugins=[plugin],
+        default_root_dir=tmpdir,
+        sync_batchnorm=True,
+    )
```

Comment:
should we check that the grads are in fp16? you could use `on_after_backward` in the lightning module to inspect the gradient dtype: https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#on-after-backward